### PR TITLE
Enable DYNAMIC_ARCH for openblas in x64 AL2 build environment

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
Enable DYNAMIC_ARCH for x64 openblas in AL2 build environment.

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4379#issuecomment-2192361269

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/ci/dockerfiles/current/build.al2.opensearch.x64.arm64.dockerfile [hunk 1]
```diff
@@ -114,6 +119,6 @@ USER $CONTAINER_USER
 WORKDIR $CONTAINER_USER_HOME
 
 # Install fpm for opensearch dashboards core
-RUN gem install dotenv -v 2.8.1 && gem install fpm -v 1.14.2
+RUN gem install dotenv -v 2.8.1 && gem install public_suffix -v 5.1.1 && gem install fpm -v 1.14.2
```
Comment (junqiu-lei) [inline docker/ci/dockerfiles/current/build.al2.opensearch.x64.arm64.dockerfile:122]: This change is used to fix the build error for:
```
Step 36/38 : RUN gem install dotenv -v 2.8.1 && gem install fpm -v 1.14.2
 ---> Running in 0f82b4ddcfdc
Successfully installed dotenv-2.8.1
Parsing documentation for dotenv-2.8.1
Installing ri documentation for dotenv-2.8.1
Done installing documentation for dotenv after 0 seconds
1 gem installed
ERROR:  Error installing fpm:
	The last version of public_suffix (>= 2.0.2, < 7.0) to support your Ruby & RubyGems was 5.1.1. Try installing it with `gem install public_suffix -v 5.1.1` and then running the current command again
	public_suffix requires Ruby version >= 3.0. The current ruby version is 2.6.0.0.
``

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/build.al2.opensearch.x64.arm64.dockerfile:122]: Add more contact that latest 6.0.0 version was released on 2024/06/17, causing the latest changed from 5 to 6 thus requires a pin on the version here.

https://rubygems.org/gems/public_suffix/versions/6.0.0

Thanks.

**Review by peterzhuamazon — APPROVED** [review]
Thanks @junqiu-lei !

**Comment by peterzhuamazon** [comment]
https://build.ci.opensearch.org/job/docker-build/5495/console

**Pull request merged**